metadata-ingestion/src/datahub/ingestion/source/unity/proxy.py

```diff
@@ -608,7 +882,9 @@ def _execute_sql_query(self, query: str, params: Sequence[Any] = ()) -> List[Row
             logger.warning(f"Failed to execute SQL query: {e}")
             return []
 
-    @cached(cachetools.FIFOCache(maxsize=100))
+    # It is enough to keep the cache size to 1, since we only process one catalog at a time
+    # We need to change this if we want to support parallel processing of multiple catalogs
+    @cached(cachetools.FIFOCache(maxsize=1))
```

Comment:
let's create a `_MAX_CONCURRENT_CATALOGS = 1` constant and put the comment there